Lib/pathlib.py

```diff
@@ -609,13 +576,20 @@ def parts(self):
         else:
             return tuple(self._tail)
 
+    def makepath(self, *args):
+        """Construct a new path object from any number of path-like objects.
+        Subclasses may override this method to customize how new path objects
+        are created from methods like `iterdir()`.
+        """
+        return type(self)(*args)
```

Comment:
```suggestion
    def makepath(self, *pathsegments):
        """Construct a new path object from any number of path-like objects.
        Subclasses may override this method to customize how new path objects
        are created from methods like `iterdir()`.
        """
        return type(self)(*pathsegments)
```

Reply:
You still need the same for joinpath as well right?